server/tso.go

```diff
@@ -163,10 +184,21 @@ func (s *Server) getRespTS(count uint32) (pdpb.Timestamp, error) {
 
 		resp.Physical = current.physical.UnixNano() / int64(time.Millisecond)
 		resp.Logical = atomic.AddInt64(&current.logical, int64(count))
+
 		if resp.Logical >= maxLogical {
-			log.Errorf("logical part outside of max logical interval %v, please check ntp time, retry count %d", resp, i)
-			time.Sleep(updateTimestampStep)
-			continue
+			// If the time window is enough, it will increase the physical time,
+			// otherwise, it will wait for updating the time window.
+			if subTimeByWallClock(s.lastSavedTime, current.physical) > updateTimestampGuard {
```

Comment:
I think `lastSavedTime` is accessed concurrently without any mutex.

Reply:
agree.